app/controllers/logs_controller.rb

```diff
@@ -147,25 +147,39 @@ def new
     @region = Region.find(params[:region_id])
     @log = Log.new
     food_type_id = @region.food_types.where('name'=>"Food")[0].id
-    @log_parts = [LogPart.new( 'food_type_id' => food_type_id)]
     @log.region = @region
     @action = 'create'
     authorize! :create, @log
     session[:my_return_to] = request.referer
     set_vars_for_form @region
-    render :new
+    if params['is_farmer_market'] == '1'
+      selected_location = Location.find(params['donor_id'])
```

Comment:
how hard would it be to change the params['is_farmer_market'] to return a boolean of `true`/`false`, so line 155 could be `params['is_farmer_market']`